src/core/qgseventtracing.cpp

```diff
@@ -0,0 +1,98 @@
+/***************************************************************************
+  qgseventtracing.cpp
+  --------------------------------------
+  Date                 : October 2019
+  Copyright            : (C) 2019 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgseventtracing.h"
+
+#include <QFile>
+#include <QThread>
+
+/// @cond PRIVATE
+
+bool QgsEventTracing::sIsTracing = false;
+QElapsedTimer QgsEventTracing::sTracingTimer;
```

Comment:
@m-kuhn will have a fit if you merge these :rofl:

Reply:
hmm... is there anything wrong with that code? The static variables are not QObject based or dependent on any other QGIS stuff, so it should be all safe...

Reply:
Non-POD global statics are somewhere in the gray zone of being reliable constructs and happened to crash QGIS on exit. That's why I took the time to kill many of them just the last days.

https://github.com/KDE/clazy/blob/master/docs/checks/README-non-pod-global-static.md

Reply:
okay thanks for the pointer - changed to use Q_GLOBAL_STATIC